owl2nl/src/main/java/org/aksw/owl2nl/Optimizer.java

```diff
@@ -60,12 +66,149 @@ else if ((list.get(i)).Key.equals("CC")) {
+                        for (int i = Integer.parseInt(verbList.get(2).Key) + 1; i < list.size(); i++) {
+                            finalText.add(list.get(i).Value);
+                        }
+                    }
+                    else{
+                        /*implement when comma comes after cc*/
+                        for (int i = 0; i <= ccList.get(0); i++) {
+                            finalText.add(list.get(i).Value);
+                        }
+                        for (int i = Integer.parseInt(verbList.get(1).Key) + 1; i <= commaList.get(0); i++) {
+                            finalText.add(list.get(i).Value);
+                        }
+                        for (int i = Integer.parseInt(verbList.get(2).Key) + 1; i < list.size(); i++) {
+                            finalText.add(list.get(i).Value);
+                        }
+                    }
+                }
+
             }
-            text = String.join(" ", finalText);
+            /*connecting comma and cc and trying to generalize*/
+            else if(ccList.size()>=1 && verbList.size()>=2 && commaList.size()>=1){
+                int loopSize=0;
+                int ccListPointer=0;
+                int commaListPointer=0;
+                loopSize=ccList.size()+commaList.size();
+                //determining loop size. this logic is not correct. correct it by summing both size.
+                //if(commaList.size()>0){
+                    //if (commaList.size()>ccList.size()){
+                    //    loopSize=commaList.size();
+                    //}
+                    //else{
+                     //   loopSize=ccList.size();
+                   // }
+
+                for (int i=0;i<loopSize;i++){
+                        if (ccListPointer<ccList.size() && commaListPointer<commaList.size()){
+                            if (commaList.get(commaListPointer)<ccList.get(ccListPointer)){
+                                combinedCcCommaList.add(commaList.get(commaListPointer));
+                                commaListPointer++;
+                            }
+                            else{
+                                combinedCcCommaList.add(ccList.get(ccListPointer));
+                                ccListPointer++;
+                            }
+                        }
+                        else if (ccListPointer>=ccList.size()){
+                            combinedCcCommaList.add(commaList.get(commaListPointer));
+                            commaListPointer++;
+                        }
+                        else if (commaListPointer>=commaList.size()){
+                            combinedCcCommaList.add(ccList.get(ccListPointer));
+                            ccListPointer++;
+                        }
+                }
+                    /*check whether the verbs are same or not*/
+                for(int j=1;j<verbList.size();j++){
+                    if(!(verbList.get(0).Value.equals(verbList.get(j).Value))){
```

Comment:
the comparison will only be with the first verb not all(each other).

Reply:
This is probably because we are only considering cases, where all the verbs used are the same. The logic for other cases has not been implemented I believe.

Reply:
yes, that is right Aparna.